src/tests/Equinor.ProCoSys.IPO.Command.Tests/EventHandlers/HistoryEvents/IpoUnAcceptedEventHandlerTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using Equinor.ProCoSys.IPO.Command.EventHandlers.HistoryEvents;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.HistoryAggregate;
+using Equinor.ProCoSys.IPO.Domain.Events;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Equinor.ProCoSys.IPO.Command.Tests.EventHandlers.HistoryEvents
+{
+    [TestClass]
+    public class IpoUnAcceptedEventHandlerTests
+    {
+        private Mock<IHistoryRepository> _historyRepositoryMock;
+        private IpoUnAcceptedEventHandler _dut;
+        private History _historyAdded;
+
+        [TestInitialize]
+        public void Setup()
+        {
+            // Arrange
+            _historyRepositoryMock = new Mock<IHistoryRepository>();
+            _historyRepositoryMock
+                .Setup(repo => repo.Add(It.IsAny<History>()))
+                .Callback<History>(history =>
+                {
+                    _historyAdded = history;
+                });
+
+            _dut = new IpoUnAcceptedEventHandler(_historyRepositoryMock.Object);
+        }
+
+        [TestMethod]
+        public void Handle_ShouldAddIpoUnAcceptedHistoryRecord()
+        {
+            // Arrange
+            Assert.IsNull(_historyAdded);
+
+            // Act
+            var objectGuid = Guid.NewGuid();
+            var plant = "TestPlant";
+            _dut.Handle(new IpoUnAcceptedEvent(plant, objectGuid), default);
+
+            // Assert
+            Assert.IsNotNull(_historyAdded);
+            Assert.AreEqual(plant, _historyAdded.Plant);
+            Assert.AreEqual(objectGuid, _historyAdded.ObjectGuid);
+            Assert.IsNotNull(_historyAdded.Description);
+            Assert.AreEqual(EventType.IpoUnaccepted, _historyAdded.EventType);
+            Assert.AreEqual("IPO", _historyAdded.ObjectType);
+        }
+    }
+}
```

Comment:
I miss the test: Handle_ShouldSave() with the Assert
_unitOfWorkMock.Verify(u => u.SaveChangesAsync(default), Times.Once);

The only  place I see this test is in DeleteAttachmentCommandHandlerTests. All Handlers handling a command should probably have such test